### ISSUE TITLE
HTTPProxy: Validate literal host rewrites are valid hostnames

### ISSUE THREAD

**Issue by sunjayBhatia** [projectcontour/contour#5788]
We should validate that when using https://github.com/projectcontour/contour/blob/bbccbffc5be5f9ccf9c42b2153ae4339a55b2b88/apis/projectcontour/v1/httpproxy.go#L1270-L1274 and rewriting the Host header with a literal hostname (not dynamic hostname that comes from another header) that is is a valid value so we do not pass invalid hostnames to Envoy to rewrite and cause issues in rejected config or in the request that gets to the backend.

              hm, might be good to do in a follow up but we should maybe be validating this doesn't get through to Envoy since this is a completely invalid hostname

_Originally posted by @sunjayBhatia in https://github.com/projectcontour/contour/pull/5678#discussion_r1340636536_

**Comment by DHRUVKHANDELWAL00** [comment]
Hey can i try this? can you please assign me?

**Comment by sunjayBhatia** [comment]
Yep @DHRUVKHANDELWAL00 assigned! Please let us know if you need any assistance on this, thanks for taking it on 👍🏽

**Comment by satyazzz123** [comment]
> We should validate that when using
> 
> https://github.com/projectcontour/contour/blob/bbccbffc5be5f9ccf9c42b2153ae4339a55b2b88/apis/projectcontour/v1/httpproxy.go#L1270-L1274
> 
> and rewriting the Host header with a literal hostname (not dynamic hostname that comes from another header) that is is a valid value so we do not pass invalid hostnames to Envoy to rewrite and cause issues in rejected config or in the request that gets to the backend.
> ```
>           hm, might be good to do in a follow up but we should maybe be validating this doesn't get through to Envoy since this is a completely invalid hostname
> ```
> 
> _Originally posted by @sunjayBhatia in [#5678 (comment)](https://github.com/projectcontour/contour/pull/5678#discussion_r1340636536)_

@sunjayBhatia Do we need to create a function for `type HeadersPolicy struct{}` for validating the hostnames? can you guide me about what should be my exact approach. Thank you.

**Comment by Garvit-77** [comment]
Hey @DHRUVKHANDELWAL00 are you still working on this issue , if not @sunjayBhatia could i give this a try

**Comment by Goutham-Annem** [comment]
Checking in here too, similar to the comment above — @DHRUVKHANDELWAL00 are you still working on this? It's been about 8 months since the last activity and #7119. Happy to pick it up if not, just want to avoid duplicating effort.
